internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -578,6 +598,40 @@ func resourceDatabricksWorkspaceCreateUpdate(d *pluginsdk.ResourceData, meta int
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
+	if defaultStorageFirewallEnabledRaw {
+		accessConnectorProperties := workspaces.WorkspacePropertiesAccessConnector{}
+		accessConnectorIdRaw := d.Get("access_connector_id").(string)
+		accessConnectorId, err := accessconnector.ParseAccessConnectorID(accessConnectorIdRaw)
+
+		if err != nil {
+			return fmt.Errorf("parsing Access Connector ID %s: %+v", accessConnectorIdRaw, err)
+		}
+
+		accessConnector, err := acClient.Get(ctx, *accessConnectorId)
+		if err != nil {
+			return fmt.Errorf("retrieving Access Connector %s: %+v", accessConnectorId.AccessConnectorName, err)
+		}
+
+		if accessConnector.Model.Identity != nil {
+
+			accIdentityId := ""
+			for raw := range accessConnector.Model.Identity.IdentityIds {
+				id, err := commonids.ParseUserAssignedIdentityIDInsensitively(raw)
+				if err != nil {
+					return fmt.Errorf("parsing %q as a User Assigned Identity ID: %+v", raw, err)
+				}
+				accIdentityId = id.ID()
```

Comment:
We probably want to break here?

```suggestion
				accIdentityId = id.ID()
				break
```